src/video/aml.c

```diff
@@ -145,31 +163,43 @@ void aml_cleanup() {
   }
 
   codec_close(&codecParam);
+  free(frame_buffer);
 }
 
 int aml_submit_decode_unit(PDECODE_UNIT decodeUnit) {
+
+  if (decodeUnit->fullLength > DECODER_BUFFER_SIZE) {
```

Comment:
Can you update this to use the new `ensure_buf_size()` helper?

See https://github.com/moonlight-stream/moonlight-embedded/commit/32e87994cdf3aa4bdad1cdf8d6a568016412c21c